### PR TITLE
[ADAM-1061] Clean up attributes regex and denormalized fields

### LINKED ISSUE #1061: scala.MatchError RegExp  does not catch colons in value part properly
line https://github.com/bigdatagenomics/adam/blob/b29d20480001be20e0be55e828a147b5f222ce42/adam-core/src/main/scala/org/bdgenomics/adam/util/AttributeUtils.scala#L31 
with 
`val attrRegex = RegExp("([^:]{2,4}):([AifZHB]):([cCiIsSf]{1},)?(.*)")`

does not handle properly alignmentrecords with attributes like
`OQ:Z:C55/15D:::::::.7GFFAFDA442.40F=AGHHE`
ie. have colons in the value part

some problematic reads are contained in gatk bundle file  CEUTrio.HiSeq.WGS.b37.NA12878.bam
```
scala> BamWriter.adamSAMSave( "output.bam", bam.sequences, bam.recordGroups , true, true ,false)
2016-06-30 17:01:41 ERROR Utils:95 - Aborting task
scala.MatchError: Z:C, (of class java.lang.String)
	at org.bdgenomics.adam.util.AttributeUtils$.createAttribute(AttributeUtils.scala:92)
	at org.bdgenomics.adam.util.AttributeUtils$.parseAttribute(AttributeUtils.scala:74)
	at org.bdgenomics.adam.util.AttributeUtils$$anonfun$parseAttributes$2.apply(AttributeUtils.scala:61)

```

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1061 and #483.

### CODE REVIEW

**Comment by fnothaft** [comment]
Tacked on one more commit (f8aeaf8) that cleans up the SAMRecordConverter class, which had not gotten cleaned up previously.

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/SAMRecordConverter.scala [hunk 1]
```diff
@@ -35,7 +35,42 @@ import org.bdgenomics.adam.util.AttributeUtils
...
+private[adam] class SAMRecordConverter extends Serializable with Logging {
+
+  /**
+   * Returns true if a tag should not be kept in the attributes field.
+   *
+   * The SAM/BAM format supports attributes, which is a key/value pair map. In
+   * ADAM, we have promoted some of these fields to "primary" fields, so that we
+   * can more efficiently access them. These include the MD tag, which describes
+   * substitutions against the reference; the OQ tag, which describes the
+   * original read base qualities; and the OP and OC tag, which describe the
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/SAMRecordConverter.scala]: `OP and OC tag`&rarr; `OP and OC tags`

**Comment by heuermh** [comment]
LGTM

**Comment by fnothaft** [comment]
Rebased and addressed review comment.

**Comment by heuermh** [comment]
Thank you, @fnothaft!

**Pull request merged**